build/gen_stub.php

```diff
@@ -629,26 +662,332 @@ private function getFlagsAsString(): string
+        }
+
+        return $code;
+    }
+
+    private function getFlagsAsString(): string
+    {
+        $flags = "ZEND_ACC_PUBLIC";
+        if ($this->flags & Class_::MODIFIER_PROTECTED) {
+            $flags = "ZEND_ACC_PROTECTED";
+        } elseif ($this->flags & Class_::MODIFIER_PRIVATE) {
+            $flags = "ZEND_ACC_PRIVATE";
+        }
+
+        if ($this->flags & Class_::MODIFIER_STATIC) {
+            $flags .= "|ZEND_ACC_STATIC";
+        }
+
+        if ($this->isDeprecated) {
+            $flags .= "|ZEND_ACC_DEPRECATED";
+        }
+
+        return $flags;
     }
 }
 
 class ClassInfo {
     /** @var Name */
     public $name;
+    /** @var int */
+    public $flags;
+    /** @var bool */
+    public $isClass;
+    /** @var bool */
+    public $isDeprecated;
+    /** @var bool */
+    public $isStrictProperties;
+    /** @var Name[] */
+    public $extends;
+    /** @var Name[] */
+    public $implements;
+    /** @var PropertyInfo[] */
+    public $propertyInfos;
     /** @var FuncInfo[] */
     public $funcInfos;
 
-    public function __construct(Name $name, array $funcInfos) {
+    /**
+     * @param Name[] $extends
+     * @param Name[] $implements
+     * @param PropertyInfo[] $propertyInfos
+     * @param FuncInfo[] $funcInfos
+     */
+    public function __construct(
+        Name $name,
+        int $flags,
+        bool $isClass,
+        bool $isDeprecated,
+        bool $isStrictProperties,
+        array $extends,
+        array $implements,
+        array $propertyInfos,
+        array $funcInfos
+    ) {
         $this->name = $name;
+        $this->flags = $flags;
+        $this->isClass = $isClass;
+        $this->isDeprecated = $isDeprecated;
+        $this->isStrictProperties = $isStrictProperties;
+        $this->extends = $extends;
+        $this->implements = $implements;
+        $this->propertyInfos = $propertyInfos;
         $this->funcInfos = $funcInfos;
     }
+
+    public function getRegistration(): string
+    {
+        $params = ["class_entry"];
+        foreach ($this->extends as $extends) {
+            $params[] = "class_entry_$extends";
+        }
+        foreach ($this->implements as $implements) {
+            $params[] = "class_entry_$implements";
+        }
+        foreach ($this->propertyInfos as $property) {
+            $type = $property->type;
+            if ($type === null) {
+                continue;
+            }
+
+            $representableType = $type->tryToRepresentableType();
+            if ($representableType && $representableType->classType) {
+                $params[] = "class_entry_" . $representableType->classType->toEscapedName();
+            }
+        }
+        $params = array_unique($params);
+
+        $code = "#define register_class_{$this->name}(";
+        $code .= implode(", ", $params);
+        $code .= ") \\\n";
+
+        $code .= "{ \\\n";
+        $code .= "\tzend_class_entry ce; \\\n \\\n";
```

Comment:
I noticed that the generated code fails for classes which define the `create_object` handler, because they need the `ce` variable. I don't know what I should do exactly. My idea is to also require this to be passed to the macro as an argument. Does it sound reasonable?

Reply:
Shouldn't it be possible to just set create_object afterwards, on the already registered CE?

Reply:
Ah, I see you're already doing that ^^

Reply:
I just got confused yesterday, mostly because I couldn't run the tests locally. The above issue happened with transliterator, but I only had to change the `->` to `.` in order to fix my issue.